Clean/Gadgets/Keccak/Chi.lean

```diff
@@ -0,0 +1,110 @@
+import Clean.Types.U64
+import Clean.Circuit.Loops
+import Clean.Gadgets.Xor.Xor64
+import Clean.Gadgets.And.And64
+import Clean.Gadgets.Not.Not64
+import Clean.Gadgets.Keccak.KeccakState
+import Clean.Specs.Keccak256
+
+namespace Gadgets.Keccak256.Chi
+variable {p : ℕ} [Fact p.Prime] [Fact (p > 512)]
+open Gadgets.Keccak256 (KeccakState)
+open Bitwise (not64)
+open Not (not64_bytewise not64_bytewise_value)
+
+def main (state : Var KeccakState (F p)) : Circuit (F p) (Var KeccakState (F p)) :=
+  Vector.mapFinRangeM 25 fun i => do
+    let state_not ← subcircuit Not.circuit (state.get (i + 5))
+    let state_and ← subcircuit And.And64.circuit ⟨state_not, state.get (i + 10)⟩
+    subcircuit Xor.circuit ⟨state.get i, state_and⟩
```

Comment:
this gadget, and its fairly easy soundness proof, is the main achievement of this PR